bootspec/src/generation.rs

```diff
@@ -33,6 +33,17 @@ impl Generation {
     }
 }
 
+impl TryInto<v1::GenerationV1> for Generation {
+    type Error = String;
+
+    fn try_into(self) -> Result<v1::GenerationV1, Self::Error> {
+        match self {
+            Self::V1(gen) => Ok(gen),
+            _ => return Err(format!("Invalid schema version, expected version 1, got: {}", self.version()).to_string())
```

Comment:
I like the idea, but I don't like the fact we'd need a separate impl block for each generation. I think switching to `TryFrom` will make this point moot, though.